src/manage_sql_nvts_common.c

```diff
@@ -0,0 +1,616 @@
+                 "   AND c_prefs.pref_id = new_pref_matches.pref_id"
+                 "  JOIN configs%s ON configs%s.id = c_prefs.config"
+                 " WHERE c_prefs.name != new_name;",
+                 modification_time,
+                 trash ? "_trash" : "",
+                 trash ? "_trash" : "",
+                 trash ? "_trash" : "",
+                 trash ? "_trash" : "");
+
+  while (next (&prefs))
+    {
+      resource_t preference;
+      const char *old_name, *new_name, *config_id, *new_pref_name;
+      gchar *quoted_new_name, *quoted_new_pref_name;
+
+      preference = iterator_int64 (&prefs, 0);
+      old_name = iterator_string (&prefs, 1);
+      new_name = iterator_string (&prefs, 2);
+      config_id = iterator_string (&prefs, 3);
+      new_pref_name = iterator_string (&prefs, 4);
+
+      g_message ("Preference '%s' of %sconfig %s changed to '%s'",
+                 old_name,
+                 trash ? "trash " : "",
+                 config_id,
+                 new_name);
+
+      quoted_new_name = sql_quote (new_name);
+      quoted_new_pref_name = sql_quote (new_pref_name);
+
+      sql ("UPDATE config_preferences%s"
+           " SET name = '%s', pref_name = '%s'"
+           " WHERE id = %llu",
+           trash ? "_trash " : "",
+           quoted_new_name,
+           quoted_new_pref_name,
+           preference);
+
+      g_free (quoted_new_name);
+      g_free (quoted_new_pref_name);
+    }
+
+  sql_commit ();
+
+  cleanup_iterator (&prefs);
+}
+
+/**
+ * @brief Set the NVT update check time in the meta table.
+ *
+ * @param[in]  count_new       Number of new VTs with current update.
+ * @param[in]  count_modified  Number of modified VTs with current update.
+ */
+void
+set_nvts_check_time (int count_new, int count_modified)
+{
+  if (sql_int ("SELECT NOT EXISTS (SELECT * FROM meta"
+               "                   WHERE name = 'nvts_check_time')"))
+    sql ("INSERT INTO meta (name, value)"
+         " VALUES ('nvts_check_time', m_now ());");
+  else if (sql_int ("SELECT value = '0' FROM meta"
+                    " WHERE name = 'nvts_check_time';"))
+    sql ("UPDATE meta SET value = m_now ()"
+         " WHERE name = 'nvts_check_time';");
+  else
+    {
+      if (count_new > 0)
+        event (EVENT_NEW_SECINFO, "nvt", 0, 0);
+
+      if (count_modified > 0)
+        event (EVENT_UPDATED_SECINFO, "nvt", 0, 0);
+
+      sql ("UPDATE meta SET value = m_now ()"
+           " WHERE name = 'nvts_check_time';");
+    }
+}
+
+/**
+ * @brief Handles database state initialization before processing NVTs.
+ *
+ * @param[in] rebuild  Whether we're rebuilding the tables.
+ */
+void
+handle_nvts_database_state (int rebuild) {
+  if (rebuild) {
+      sql("DROP TABLE IF EXISTS vt_refs_rebuild;");
+      sql("DROP TABLE IF EXISTS vt_severities_rebuild;");
+      sql("DROP TABLE IF EXISTS nvt_preferences_rebuild;");
+      sql("DROP TABLE IF EXISTS nvts_rebuild;");
+      create_tables_nvt("_rebuild");
```

Comment:
I would add a new line here
```suggestion

      create_tables_nvt("_rebuild");
```